src/monster_group.cpp

```diff
@@ -0,0 +1,57 @@
+#include "monster_group.h"
+
+#include <algorithm>
+
+static bool cmp_monster_by_update( const monster &lhs, const monster &rhs ) {
+    return lhs.next_overmap_move() < rhs.next_overmap_move();
+}
+
+void monster_group::add_monster( monster &new_monster ) {
+    // Reset the strongest stimulus since this new monster might not have received it.
```

Comment:
On the other hand, the newest monster might simply follow its group. Heuristically, it might be a better solution for now to keep the stimulus, but decrease the interest in the stimulus.